python/scripts/cirkopt.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+
+# Set up path relative to python root folder so we can find the other packages
+import sys
+import os.path
+import argparse
+import logging
+from logging import DEBUG, INFO, info, WARNING, error
+from numpy import arange
+
+PYTHON_SCRIPTS_DIRECTORY: str = os.path.dirname(os.path.abspath(__file__))
+PYTHON_DIRECTORY: str = os.path.abspath(os.path.join(PYTHON_SCRIPTS_DIRECTORY, ".."))
+LIBERATE_DIRECTORY: str = os.path.abspath(os.path.join(PYTHON_DIRECTORY, "../liberate"))
+sys.path.append(PYTHON_DIRECTORY)
+
+# These imports rely on changed sys.path
+from scripts.single_param_sweep import (  # pylint: disable=wrong-import-position
+    main as sweep_param,
+)
+from src.single_param_sweep import Param  # pylint: disable=wrong-import-position
+
+# Basically a copy of this blog post [1].
+# [1]: https://chase-seibert.github.io/blog/2014/03/21/python-multilevel-argparse.html
+class Cirkopt:
+    def __init__(self):
+        parser = argparse.ArgumentParser(
+            description="SPICE circuit optimizer",
+            usage="""
+cirkopt <command> [<args>]
+
+The most commonly used commands are:
+explore     Generate plots showing search space
+search      Find an optimal design""",
+        )
+        parser.add_argument("command", help="Sub command to run")
+        # parse_args defaults to [1:] for args, but you need to
+        # exclude the rest of the args too, or validation will fail
+        args = parser.parse_args(sys.argv[1:2])
+        if not hasattr(self, args.command):
+            error("Unrecognized command")
+            parser.print_help()
+            sys.exit()
+        # use dispatch pattern to invoke method with same name
+        getattr(self, args.command)()
+
+    # pylint: disable=no-self-use
+    def explore(self):
+        parser = argparse.ArgumentParser(
+            description="Generate plots showing search space",
+        )
+        # prefixing the argument with -- means it's optional
+        parser.add_argument(
+            "--param",
+            help="Paramater to sweep, e.g.: width",
+            default="width",
+        )
+        parser.add_argument(
+            "--range",
+            help=(
+                "Range of numbers to sweep over, MATLAB style (start:increment:end)"
+                " e.g.: 200e-9:100e-9:1000e-9"
+            ),
+            default="200e-9:100e-9:1000e-9",
+        )
+        parser.add_argument(
+            "--outpin",
+            help="Name of pin to get data from, e.g.: Y",
+            default="Y",
+        )
+        parser.add_argument(
+            "--outindex",
+            help="Index of value from LDB table to show, comma separated, e.g.: 0,1",
+            default="0,1",
```

Comment:
you don't have to parse this yourself, argparse handles args with list of values. Use
```python
parser.add_argument(
            "--outindex",
            nargs=2,    
            type=int,
            help="Index of value from LDB table to show, comma separated, e.g.: 0,1",
            default=[0, 1],
)
```

Reply:
I thought that would need to be space separated, but I'll try it (not really familiar)

Reply:
I think it does have to be space separated. I don't see comma separated vs space separated being any different

Reply:
I think something more flexible than either approach would be more ideal, but I find entering a 2D index as comma separated more intuitive since it is like a coordinate. Same for the colon separated seems more intuitive since a lot of languages like matlab use colons for ranges

Reply:
Haha i disagree but I also don't really care, I'm fine with parsing them ourselves if you want